unit-tests/src/test/scala/java/util/CollectionTest.scala

```diff
@@ -0,0 +1,293 @@
+// Ported from Scala.js commit: f9fc1ae dated: 2020-03-06
+
+package org.scalanative.testsuite.javalib.util
+
+import java.{util => ju, lang => jl}
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalanative.testsuite.javalib.lang.IterableFactory
+import org.scalanative.testsuite.javalib.lang.IterableTest
+
+import scala.reflect.ClassTag
+
+import scala.scalanative.junit.utils.AssertThrows._
+import scala.scalanative.junit.utils.Utils._
+
+trait CollectionTest extends IterableTest {
```

Comment:
This file on its own is all dead code. We would need at least one concrete subclass of `CollectionTest` with a concrete factory type to make this non dead code.

So either `CollectionTest` and `TrivialImmutableCollection` should be removed from this PR (and instead introduced in a PR that actually has a least one concrete subclass), or we should add such a concrete subclass in this PR.

Reply:
I chose the second alternative and moved  AbstractCollection.scala & AbstractCollectionTest.scala
from PR #2014.  This keeps the two sets of file.scala & fileTest.scala in the same PR.  
AbstractCollectionTest implements the concrete subclass.

Reply:
PS: thank you for clearly describing the alternatives.